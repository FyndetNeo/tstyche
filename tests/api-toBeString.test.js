import { expect, test } from "@jest/globals";
import * as tstyche from "tstyche";
import { getFixtureUrl } from "./__utils__/fixtureFactory.js";
import { normalizeOutput } from "./__utils__/normalizeOutput.js";
import { spawnTyche } from "./__utils__/spawnTyche.js";

const fixtureUrl = getFixtureUrl("api-toBeString");

test("'toBeString' implementation", () => {
  expect(tstyche.expect).toHaveProperty("type.toBeString", expect.any(Function));
});

test("toBeString", async () => {
  const { exitCode, stderr, stdout } = await spawnTyche(fixtureUrl);

  expect(normalizeOutput(stdout)).toMatchSnapshot("stdout");
  expect(stderr).toMatchSnapshot("stderr");

  expect(exitCode).toBe(1);
});
