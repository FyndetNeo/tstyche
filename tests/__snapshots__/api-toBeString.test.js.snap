// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toBeString: stderr 1`] = `
"Error: The source type is 'void'.

   7 |   expect(returnsString()).type.toBeString();
   8 | 
>  9 |   expect(returnsVoid()).type.toBeString();
     |                              ^
  10 | });
  11 | 
  12 | test("is NOT string?", () => {

       at ./__typetests__/toBeString.tst.ts:9:30 ❭ is string?

Error: The source type is 'string'.

  13 |   expect(returnsVoid()).type.not.toBeString();
  14 | 
> 15 |   expect(returnsString()).type.not.toBeString();
     |                                    ^
  16 | });
  17 | 

       at ./__typetests__/toBeString.tst.ts:15:36 ❭ is NOT string?

"
`;

exports[`toBeString: stdout 1`] = `
"uses TypeScript <<version>> with ./tsconfig.json

fail ./__typetests__/toBeString.tst.ts
  × is string?
  × is NOT string?

Targets:    1 failed, 1 total
Test files: 1 failed, 1 total
Tests:      2 failed, 2 total
Assertions: 2 failed, 2 passed, 4 total
Duration:   <<timestamp>>

Ran all test files.
"
`;
